src/services/codefixes/fixUnmatchedParameter.ts

```diff
@@ -0,0 +1,96 @@
+/* @internal */
+namespace ts.codefix {
+    const deleteUnmatchedParameter = "deleteUnmatchedParameter";
+    const renameUnmatchedParameter = "renameUnmatchedParameter";
+
+    const errorCodes = [
+        Diagnostics.JSDoc_param_tag_has_name_0_but_there_is_no_parameter_with_that_name.code,
+    ];
+
+    registerCodeFix({
+        fixIds: [deleteUnmatchedParameter, renameUnmatchedParameter],
+        errorCodes,
+        getCodeActions: function getCodeActionsToFixUnmatchedParameter(context) {
+            const { sourceFile, span } = context;
+            const actions: CodeFixAction[] = [];
+            const info = getInfo(sourceFile, span.start);
+            if (info) {
+                append(actions, getDeleteAction(context, info));
+                append(actions, getRenameAction(context, info));
+                return actions;
+            }
+            return undefined;
+        },
+        getAllCodeActions: function getAllCodeActionsToFixUnmatchedParameter(context) {
+            const tagsToSignature = new Map<SignatureDeclaration, JSDocTag[]>();
+            return createCombinedCodeActions(textChanges.ChangeTracker.with(context, changes => {
+                eachDiagnostic(context, errorCodes, ({ file, start }) => {
+                    const info = getInfo(file, start);
+                    if (info) {
+                        tagsToSignature.set(info.signature, append(tagsToSignature.get(info.signature), info.jsDocParameterTag));
+                    }
+                });
+
+                tagsToSignature.forEach((tags, signature) => {
+                    if (context.fixId === deleteUnmatchedParameter) {
+                        const tagsSet = new Set(tags);
+                        changes.filterJSDocTags(signature.getSourceFile(), signature, t => !tagsSet.has(t));
+                    }
+                });
+            }));
+        }
+    });
+
+    function getDeleteAction(context: CodeFixContext, { name, signature, jsDocParameterTag }: Info) {
+        const changes = textChanges.ChangeTracker.with(context, changeTracker =>
+            changeTracker.filterJSDocTags(context.sourceFile, signature, t => t !== jsDocParameterTag));
+        return createCodeFixAction(deleteUnmatchedParameter, changes, [Diagnostics.Remove_unused_parameter_0, name.getText(context.sourceFile)], deleteUnmatchedParameter, Diagnostics.Remove_all_unused_parameters);
+    }
+
+    function getRenameAction(context: CodeFixContext, { name, signature, jsDocParameterTag }: Info) {
+        const sourceFile = context.sourceFile;
+        if (length(signature.parameters)) {

```

Comment:
I'd invert this if and early return, or just leave out the check entirely and let the one following `firstDefined` take care of it (although I can't remember if `firstDefined` throws on empty arrays.)